src/Util/extractCloudinaryProps.js

```diff
@@ -0,0 +1,37 @@
+import {Util} from "cloudinary-core";
+
+const isDefined = (props, key) => {
+  return (props[key] !== undefined && props[key] !== null);
+};
+
+/**
+ * Extracts cloudinaryProps and nonCloudinaryProps from given context+props
+ * according to given validProps
+ * @param context
+ * @param props
+ * @param validProps
+ * @returns {{cloudinaryProps, nonCloudinaryProps}}
+ */
+export default (context = {}, props = {}, validProps = {}) => {
```

Comment:
The merge does not belong here. 
It should be done outside and here you should get a list of props and a list of cloudinaryProps